### PR TITLE
Update to stubparser-3.20.2

### PR DESCRIPTION
**Pull request by smillst**
Merge with https://github.com/typetools/stubparser/pull/55.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -280,6 +280,11 @@ task maybeCloneAndBuildDependencies(type: Exec) {
         // the repository does not exist when building checker.jar.
     }
     executable 'checker/bin-devel/build.sh'
+    doLast{
```
Comment (mernst) [inline build.gradle:283]: Nit: I would add a space before `{`.

**Review by mernst — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (smillst) [reply build.gradle]: I moved this change (and added some more) to https://github.com/typetools/checker-framework/pull/4473.

**Comment by smillst** [comment]
The error message:
```
warning: Memory constraints are impeding performance; please increase max heap size (max memory = 536870912, total memory = 536870912, free memory = 2944856)
warning: Memory constraints are impeding performance; please increase max heap size (max memory = 536870912, total memory = 536870912, free memory = 14402064)
error: warnings found and -Werror specified
```

File: framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java [hunk 2]
```diff
@@ -608,7 +610,13 @@ private void parseStubUnit(InputStream inputStream) {
     if (debugAnnotationFileParser) {
       stubDebug(String.format("parsing stub file %s", filename));
     }
-    stubUnit = StaticJavaParser.parseStubUnit(inputStream);
+    ParserConfiguration configuration = new ParserConfiguration();
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java:613]: We may want to make this change everywhere we use StaticJavaParser.

Comment (mernst) [reply framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java:613]: How about merging https://github.com/typetools/checker-framework/pull/4499 first?

Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java]: I've done that.  Thanks!

**Pull request merged**